src/components/Truncate/Truncate.tsx

```diff
@@ -117,14 +120,37 @@ export class Truncate extends React.PureComponent<
     )
 
     const shouldShowTooltip = showTooltipOnTruncate && this.state.isTruncated
-    const word = this.getTruncatedContent()
+    let truncatedText
+
+    if (splitter) {
+      const str = text || children
+      const [first, second] = str.split(splitter)
+
+      truncatedText = (
+        <TruncateWithSplitterUI
+          className={`${TRUNCATED_CLASSNAMES.component} ${
+            TRUNCATED_CLASSNAMES.withSplitter
+          }`}
+        >
+          <span className={`${TRUNCATED_CLASSNAMES.firstChunk}`}>{first}</span>
+          <span className={`${TRUNCATED_CLASSNAMES.splitterChunk}`}>
+            {splitter}
+          </span>
+          <span className={`${TRUNCATED_CLASSNAMES.secondChunk}`}>
+            {second}
+          </span>
+        </TruncateWithSplitterUI>
```

Comment:
In terms of the class names here, what do you think with sticking with the `bem` approach? Perhaps, `c-c-Truncate__content__chunks` for the wrapper around chunks and `c-Truncate__content__chunks__first` for a chunk. Also, instead of `withSplitter` could we use `with-splitter` for consistency with other modifiers, like `is-visible`?